### PR TITLE
wasm: net.cidr_contains, net.cidr_intersects, and net.cidr_overlap builtins.

### PR DESCRIPTION
**Pull request by koponen-styra**
Signed-off-by: Teemu Koponen <koponen@styra.com>

### CODE REVIEW

File: wasm/src/cidr.c [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        return FALSE;
+    }
+
+    for (int i = 0; i < a->len; i++)
+    {
+        if (a->mask[i] & ~b->mask[i])
+        {
+            // b mask is shorter.
+            return FALSE;
+        }
```
Comment (tsandall) [inline wasm/src/cidr.c:104]: Is this just an optimization? I didn't work through the bit arithmetic to convince myself it was correct.

File: wasm/src/cidr.c [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+        if (a->ip[i] != (b->ip[i] & a->mask[i]))
```
Comment (tsandall) [inline wasm/src/cidr.c:106]: A comment mentioning that the mask is applied to the CIDR above (line 88?) might be helpful for our future selves. Alternatively just apply the mask to both IPs.

**Review by tsandall — APPROVED** [review]
LGTM. Mostly just questions.

File: wasm/src/cidr.c (hunk 2 above)
Comment (koponen-styra) [reply wasm/src/cidr.c:122]: Added a comment.

File: wasm/src/cidr.c (hunk 1 above)
Comment (koponen-styra) [reply wasm/src/cidr.c]: Added a comment.

**Review by tsandall — APPROVED** [review]

**Pull request merged**